From 6dc0f1f51ab4fbc907650713d8310126ffb0af3a Mon Sep 17 00:00:00 2001
From: Kai Koehne <kai.koehne@qt.io>
Date: Sun, 1 Nov 2020 21:37:48 +0100
Subject: [PATCH 07/17] CMake: Fix usage of qt_create_translation in a CTest

Commit 1b006ab3d8954 declared an input file generated by CMake as a
BYPRODUCT of the consuming file to fix a policy warning CMP0058.
Anyhow, this is technically incorrect, since the file is actually not produced,
but consumed by the lupdate command. The incorrect fix also had the
side-effect that ninja clean did remove the file, which prevents the use
of the macro in CTest.

Just remove the BYPRODUCT declaration. The file is written by
CMake, so it actually doesn't need neither to be listed in DEPENDENCIES
nor in BYPRODUCT.

Task-number: QTBUG-71354
Change-Id: I7e5677f2bbd84b8a9fffdcb73465dfae6012ceb5
Reviewed-by: Alexandru Croitor <alexandru.croitor@qt.io>
(cherry picked from commit 3101eae233f1f5031c5c814d70dedbc7986e7db8)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/linguist/Qt5LinguistToolsMacros.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/linguist/Qt5LinguistToolsMacros.cmake b/src/linguist/Qt5LinguistToolsMacros.cmake
index 2e700bba..3138aae4 100644
--- a/src/linguist/Qt5LinguistToolsMacros.cmake
+++ b/src/linguist/Qt5LinguistToolsMacros.cmake
@@ -94,7 +94,7 @@ function(QT5_CREATE_TRANSLATION _qm_files)
             COMMAND ${Qt5_LUPDATE_EXECUTABLE}
             ARGS ${_lupdate_options} "@${_ts_lst_file}" -ts ${_ts_file}
             DEPENDS ${_dependencies}
-            BYPRODUCTS ${_ts_lst_file} VERBATIM)
+            VERBATIM)
     endforeach()
     qt5_add_translation(${_qm_files} ${_my_tsfiles})
     set(${_qm_files} ${${_qm_files}} PARENT_SCOPE)
-- 
2.31.1

